src/Microsoft.NetCore.Analyzers/Core/Performance/UsePropertyInsteadOfCountMethodWhenAvailable.cs

```diff
@@ -0,0 +1,282 @@
+                }
+
+                if (isCollectionOfTInterface(invocationTarget))
+                {
+                    return true;
+                }
+
+                if (invocationTarget.TypeKind == TypeKind.Interface)
+                {
+                    if (invocationTarget.GetMembers(CountPropertyName).OfType<IPropertySymbol>().Any())
+                    {
+                        return false;
+                    }
+
+                    foreach (var @interface in invocationTarget.AllInterfaces)
+                    {
+                        if (@interface.OriginalDefinition is INamedTypeSymbol originalInterfaceDefinition &&
+                            isCollectionOfTInterface(originalInterfaceDefinition))
+                        {
+                            return true;
+                        }
+                    }
+                }
+                else
+                {
+                    foreach (var @interface in invocationTarget.AllInterfaces)
+                    {
+                        if (@interface.OriginalDefinition is INamedTypeSymbol originalInterfaceDefinition &&
+                            isCollectionOfTInterface(originalInterfaceDefinition))
+                        {
+                            if (invocationTarget.FindImplementationForInterfaceMember(@interface.GetMembers(CountPropertyName)[0]) is IPropertySymbol propertyImplementation &&
+                                !propertyImplementation.ExplicitInterfaceImplementations.Any())
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                return false;
+
+                bool isCollectionOfTInterface(ITypeSymbol type)
+                    => this.ICollectionOfTType.Equals(type.OriginalDefinition);
+            }
+
+            internal bool IsEnumerableType(ISymbol symbol)
+                => this.EnumerableType.Equals(symbol);
+        }
+
+        /// <summary>
+        /// Handler for operaction actions.
+        /// </summary>
+        private abstract class OperationActionsHandler
+        {
+            protected OperationActionsHandler(OperationActionsContext context)
+            {
+                Context = context;
+            }
+
+            public OperationActionsContext Context { get; }
+
+            internal void AnalyzeInvocationOperation(OperationAnalysisContext context)
+            {
+                var invocationOperation = (IInvocationOperation)context.Operation;
+
+                if (GetEnumerableCountInvocationTargetType(invocationOperation) is ITypeSymbol invocationTarget &&
+                    GetReplacementProperty(invocationTarget) is string propertyName)
+                {
+                    var propertiesBuilder = ImmutableDictionary.CreateBuilder<string, string>();
+                    propertiesBuilder.Add("PropertyName", propertyName);
```

Comment:
> "PropertyName"

Define a constant field in the analyzer that is shared by the fixer.

Reply:
Forgot that one.